### PR TITLE
Adding reporter which support RunKit environment

### PR DESCRIPTION
**Pull request by rpgeeganage**
With this PR:
I'm trying to utilize 
`ValueViewerSymbol`  from `@runkit/value-viewer` and display HTML output (https://github.com/fatso83/mini-mocha/issues/6)
RunKit reporter will collect logs and flush it when a single execution finished.

Hope to get feedback.
Thanks.

### CODE REVIEW

**Comment by rpgeeganage** [comment]
@fatso83 ,
When I check the `RunKit` site, it is still loading the old version (2.0.1).
see https://runkit.com/ruwan/5d02c6cb9efd8200193ac344
Is there anything we have to do?

**Comment by fatso83** [comment]
No, it's related to NPM being a dickhead company at the moment and causing havoc for people relying on them. See
https://discuss.runkit.com/t/not-found-npm-module/1601/4 and https://npm.community/t/skimdb-is-down-for-me-anyone-else/8060/7.

At the moment, no scoped packages can be updated :-/ Probably will be fixed soon, but no news when.

**Comment by rpgeeganage** [comment]
wow, that's sad.

File: lib/reporters/run_kit_reporter.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+const { ValueViewerSymbol } = require("@runkit/value-viewer");
```
Comment (fatso83) [inline lib/reporters/run_kit_reporter.js:1]: What does this _do_? I never wrapped my head around Symbols. I am also not sure how a Symbol differs from a normal string constant - apart from enurable true/false.

File: lib/reporters/run_kit_reporter.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    flush() {
+        if (this.logs.length) {
+            // we collect them and flush
+            const first = this.logs[0];
+            const messages = this.logs.map(singleLog => `<tr><td>${singleLog.message}</td></tr>`);
+            const html = `<table><thead>${first.title}</thead>${messages.join("")}</table>`;
+
+            console.log({
+                [ValueViewerSymbol]: { title: first.title.replace(/\n/g, ""), HTML: html }
```
Comment (fatso83) [inline lib/reporters/run_kit_reporter.js:45]: Could you explain this line to me? How does this work?

I am assuming `[ValueViewerSymbol]` is replaced by some magic string constant that is then picked up by RunKit's evaluator and then it uses the `HTML` field to paste that into a designated container. Yes?

**Review by fatso83 — APPROVED** [review]
Very nice work and use of (implicit) interfaces! :100: 

Just have some questions. Merge at will.

File: lib/reporters/run_kit_reporter.js (hunk 1 above)
Comment (rpgeeganage) [reply lib/reporters/run_kit_reporter.js:1]: I'm also very new to symbols. I rarely used them but as a concept I like it. 
The thing is if we define a symbol, we cannot access it without the particular symbol or call `getOwnPropertySymbols` and iterate. it gives us a guarantee that the particular value will not get overwritten. That's why I used a symbol for `metaData`. so if somebody set `this.metaData` I know my values won't get overridden.

File: lib/reporters/run_kit_reporter.js (hunk 2 above)
Comment (rpgeeganage) [reply lib/reporters/run_kit_reporter.js:45]: I checked the code from the package, it uses string if the `Symbol` is not defined in the Javascript runtime environment. https://github.com/runkitdev/value-viewer/blob/master/value-viewer.js#L8

**Comment by rpgeeganage** [comment]
@fatso83 ,
Will you be able to publish the new changes after I merge this? and can we add a guide on how to publish the package?  because I'm not sure how to publish. thank a lot.

**Comment by fatso83** [comment]
Sure. This is how to publish:
1. Decide what kind of semantic version this is: `patch` (fixes bug), `minor` (adds feature)`, `major` (breaks API)
2. Increase the version number using `npm version patch` (or minor or `major`).
3. `npm publish` (you will need to `npm login` if you are not logged in

**Comment by fatso83** [comment]
P.S. We can change the package name to be be the unscoped `mini-mocha`, if you don't oppose. Right now the RunKit issues are preventing anyone from using it.

**Comment by fatso83** [comment]
P.P.S. Just publish an update to `@fatso83/mini-mocha` first, and then we can look at changing the name

**Comment by rpgeeganage** [comment]
sure. I'm fine with it. Now the runkit has the latest version. I think they fixed it.
Do you think this package will get included in Sinon ?

**Comment by fatso83** [comment]
I don't know. I am not super-sure if it fits in ... I can ask, but all the other packages are either 
- stubbing libs
- helpers for stubbing libs
- assertions
- some mix of the above (like `referee-sinon`)

Not sure where `mini-mocha` would fit in, but it's certainly related to the whole testing domain :-)

**Comment by rpgeeganage** [comment]
ok. it makes sense.

**Comment by fatso83** [comment]
Did you want to publish or should I do it?

**Comment by rpgeeganage** [comment]
if you are ok with the current open PR, we can merge it and publish. Please go ahead and publish. thanks. 😄

**Pull request merged**